tests/components/devolo_home_network/test_diagnostics.py

```diff
@@ -32,14 +33,4 @@ async def test_entry_diagnostics(
         entry_dict["data"][key] = REDACTED
 
     result = await get_diagnostics_for_config_entry(hass, hass_client, entry)
-    assert result == {
-        "entry": entry_dict,
-        "device_info": {
-            "mt_number": DISCOVERY_INFO.properties["MT"],
-            "product": DISCOVERY_INFO.properties["Product"],
-            "firmware": DISCOVERY_INFO.properties["FirmwareVersion"],
-            "device_api": True,
-            "plcnet_api": True,
-            "features": DISCOVERY_INFO.properties["Features"].split(","),
-        },
-    }
+    assert result == {"entry": entry_dict, "device_info": snapshot}
```

Comment:
```suggestion
    assert result == snapshot
```

Reply:
I my tests, that's not working because the entry_dict contains the entry_id and that changes on every testrun. Am I making a mistake here?

Reply:
One option is to set the `entry_id` when you create the mock entry:
`MockConfigEntry(..., entry_id="123456")`

Another option is to adjust `async_get_config_entry_diagnostics`
https://github.com/home-assistant/core/blob/c507ca1e665abffd9b8986f0974f74d35cde1653/homeassistant/components/sfr_box/diagnostics.py#L23-L27

Reply:
ok, I understood my options. I will split the diagnostics platform from this PR as well so that I can think about the options in detail.